Microsoft.Azure.CosmosRepository/src/IRepository.cs

```diff
@@ -219,5 +220,22 @@ public interface IRepository<TItem> where TItem : IItem
         /// <returns>A <see cref="ValueTask"/> representing the asynchronous exists operation.</returns>
         ValueTask<bool> ExistsAsync(Expression<Func<TItem, bool>> predicate,
             CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Offers a load more paging implementation for infinite scroll scenarios.
+        /// Allows for efficient paging making use of cosmos DBs continuation tokens, making this implementation cost effective.
+        /// </summary>
+        /// <param name="lastPage">The last page previous query, defaults to 0 for the first page</param>
+        /// <param name="predicate">A filter criteria for the paging operation, if null it will get all <see cref="IItem"/>s</param>
+        /// <param name="pageSize">The size of the page to return from cosmos db.</param>

```

Comment:
Don't forget that MaxItemCount is only a maximum - Cosmos may send back less items. I think there's two possible options here:

1. Expose this to the user by renaming `pageSize` to `maxPageSize` and clearly documenting it. It then becomes the user's responsibility to call ScrollAsync repeatedly until they get enough items to have a full page.
2. Loop internally yourself until you get the requested pageSize.

I suspect that for most actual real-world usages, (2) would be more useful - but I'm not sure how high-level/low-level an API you're looking to create here.

Reply:
I had assumed that a user would expect fewer items if there is not enough data in cosmos to satisfy the page size, was this what you meant, or did you mean we may have to do multiple calls to `ReadNextAsync()` ?

Reply:
So the way I'm reading [the docs](https://docs.microsoft.com/en-us/azure/cosmos-db/sql/sql-query-pagination), Cosmos may return less than MaxItemCount even if there are enough items in the resultset:

> The number of items returned per query execution will always be less than or equal to MaxItemCount. However, it is possible that other criteria might have limited the number of results the query could return. If you execute the same query multiple times, the number of pages might not be constant. For example, if a query is throttled there may be fewer available results per page, which means the query will have additional pages. In some cases, it is also possible that your query may return an empty page of results.

Now, if the main usage for this API is for a paging UI (I suspect that's the case), then the user wants to always fill out e.g. 25 items in each page (except the last). The way the code is currently structured, I think they'd need to call ScrollAsync repeatedly until they have the 25 items (this also complicates things for the page count, BTW). My (2) proposal above would be to do that loop inside ScrollAsync, guaranteeing that it always returns 25 items, unless there are no more results at all.

Note also that Cosmos may return 0 items as well, although there are items available. Keep this in mind with regards to detecting the end of the query resultset.

Reply:
Ah okay this makes sense now, yes look's as if we will need to be more intelligent here, I will update this shortly and push the changes.

Reply:
Hi @roji finally got around to updating this, think this should do the trick now 😃